dartagnan/src/main/java/com/dat3m/dartagnan/encoding/EncodingContext.java

```diff
@@ -261,36 +248,40 @@ public Formula lastValue(MemoryObject base, int offset, int size) {
         return formulaManager.getBitvectorFormulaManager().makeVariable(size, name);
     }
 
-    public BooleanFormula equal(Formula left, Formula right) {
-        //FIXME: We should avoid the automatic conversion, or standardize what conversion we expect.
-        // For example, when encoding rf(w, r), we always want to convert the store value type to the read value type
-        // for otherwise, we might get an under-constrained value for val(r) (e.g., its upper bits might be unconstrained,
-        // if we truncate it to smaller size of the store value).
+    public enum ConversionMode {
+        NO,
+        LEFT_TO_RIGHT,
+        RIGHT_TO_LEFT,
+    }
+
+    public BooleanFormula equal(Formula left, Formula right, ConversionMode cMode) {
+        if (cMode == ConversionMode.LEFT_TO_RIGHT) {
+            return equal(right, left, ConversionMode.RIGHT_TO_LEFT);
+        } else if (cMode == ConversionMode.NO) {
+            new EncodingHelper(formulaManager).checkEqualTypes(left, right);
```

Comment:
Either this just returns a Boolean that gets lost or I am missing something

Reply:
You are right. The checking method used to throw an exception but then I moved it and changed it to return a bool and forgot to change the callsite.